src/twisted/conch/scripts/ckeygen.py

```diff
@@ -202,45 +203,64 @@ def _defaultPrivateKeySubtype(keyType):
         return "PEM"
 
 
-def printFingerprint(options):
-    if not options["filename"]:
+def _getKeyOrDefault(options, input_collector=None):
+    """
+    If C{options["filename"]} is None, prompt the user to enter a path
+    or attempt to set it to .ssh/id_rsa
+    """
+    if input_collector is None:
+        input_collector = input
+    filename = options["filename"]
+    if not filename:
```

Comment:
Suggestion to implement these comments (1/2):

```suggestion
from typing import Dict, Optional, Callable
def _getKeyOrDefault(options: Dict[str, Optional[str]], inputCollector: Optional[Callable[[str],str]]=None) -> str:
    """
    If C{options["filename"]} is None, prompt the user to enter a path
    or attempt to set it to .ssh/id_rsa
    """
    if inputCollector is None:
        inputCollector = input
    filename = options["filename"]
    if not filename:
```

Reply:
Added typing